Assets/Game/Singletons/SpellMap.cs

```diff
@@ -39,16 +47,45 @@ void Start()
     void Update()
     {
         //Update cooldown timers
-        for (uint i = 0; i < cooldownTimers.Length; ++i)
-            for (uint j = 0; j < cooldownTimers[i].Length; ++j)
-                cooldownTimers[i][j] -= Time.deltaTime;
+        for (int playerNum = 0; playerNum < cooldownTimers.Length; ++playerNum)
+            for (int spellType = 0; spellType < cooldownTimers[playerNum].Length; ++spellType)
+                DecrementCooldown(Time.deltaTime, playerNum, (SpellType)spellType);
     }
 
-    private bool SpellReady(int playerNumber, SpellType spellType) { return cooldownTimers[playerNumber][(uint)spellType] <= 0.0f; }
+    private bool SpellReady(int playerNumber, SpellType spellType) { return cooldownTimers[playerNumber][(int)spellType] <= 0.0f; }
+
     private void RestartCooldown(int playerNumber, SpellType spellType)
+    {
+        SetCooldown(GetInitialCooldown(playerNumber, spellType), playerNumber, spellType);
+    }
+
+    private void DecrementCooldown(float deltaTime, int playerNumber, SpellType spellType)
```

Comment:
```suggestion
    private void DecreaseCooldown(float deltaTime, int playerNumber, SpellType spellType)
```
I would argue that decrement suggests that you are decreasing the cooldown by a uniform amount each time.